pkg/volume/csi/csi_plugin.go

```diff
@@ -801,11 +791,67 @@ func (p *csiPlugin) skipAttach(driver string) (bool, error) {
 	return false, nil
 }
 
+// supportsVolumeMode looks up CSIDriver object associated with the
+// driver name to determine if the driver supports the given volume
+// mode. An error indicates that it isn't supported and explains why.
+func (p *csiPlugin) supportsVolumeMode(driver string, volumeMode storage.CSIVolumeMode) error {
+	supportedModes, err := p.lookupVolumeModes(driver)
+	switch {
+	case err != nil:
+		return err
+	case containsVolumeMode(supportedModes, volumeMode):
+		// Explicitly listed.
+		return nil
+	case volumeMode == storage.PersistentVolumeMode && len(supportedModes) == 0:
```

Comment:
If we gate this entire function on CSIInlineVolume, then we don't need this case because the field will be defaulted when we read the CSIDriver object.  This way we don't encode our own defaulting logic outside of API defaulting.

Reply:
No, not quite. This case is also hit if:
- the feature is enabled
- a driver doesn't have a CSIDriver object
- the volume mode is "persistent"

In that case it's okay to proceed with the volume. If we removed this case, we would continue with the error cases.

Reply:
There's a difference between "CSIDriver object doesn't exist" and "CSIDriver does exist but is missing the field".  When the feature is enabled, the CSIDriver object should also have volumeModes set, either due to defaulting or the user explicitly setting the value.  If the field is not set, then that perhaps indicates that the feature is disabled on the apiserver, which is a configuration error.  I think in that case, we want to return an error here instead of assuming persistent mode

Reply:
Okay. To implement this distinction, I merged `lookupVolumeModes` into `supportsVolumeMode` because otherwise that function wouldn't have enough information. It's less code this way, too.

In several test cases for persistent volumes the missing field had to be added, because defaulting is not active when we fake the API server. I also added some more test cases for "missing mode" or "wrong mode for persistent volume".